ext/vscode/src/commands/loginCli.ts

```diff
@@ -1,13 +1,20 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
+import * as vscode from 'vscode';
 import { IActionContext } from '@microsoft/vscode-azext-utils';
 import { TelemetryId } from '../telemetry/telemetryId';
 import { createAzureDevCli, onAzdLoginAttempted } from '../utils/azureDevCli';
 import { executeAsTask } from '../utils/executeAsTask';
 import { getAzDevTerminalTitle } from './cmdUtil';
 
 export async function loginCli(context: IActionContext, shouldPrompt: boolean = true): Promise<void> {
+    if (vscode.workspace.getConfiguration('azure-dev').get<boolean>('auth.useIntegratedAuth', false)) {

```

Comment:
I think `azd auth login` should still be allowed and should work even from the extension; if an older version of AZD (without the ability to get tokens from VSCode) is present alongside the newer extension this could basically just disable the login command.

Reply:
Sure - we can do that.  I'll change `azd auth login` to try to do a smart thing when using external auth.